esp/esdllib/esdl_script.cpp

```diff
@@ -688,8 +688,9 @@ class CEsdlTransformOperationMySqlCall : public CEsdlTransformOperationBase
         }
 
         int type = 0;
-        while((type = xpp.next()) != XmlPullParser::END_DOCUMENT)
```

Comment:
This doesn't look right.  Previously END_DOCUMENT would have finished the loop, and continued, now it will return.

Reply:
The pull parser will throw an exception if it reaches the end of a document before an expected end tag. The END_DOCUMENT condition was never correct. The loop should probably exit on END_TAG, instead of END_DOCUMENT, and the inner switch can be simplified to just a test for START_TAG.

Reply:
I have changed loop terminating conditions from END_DOCUMENT, which is unexpected, to END_TAG. I have also created https://track.hpccsystems.com/browse/HPCC-29138 to address making the loops more robust in the event that future coding errors make END_DOCUMENT possible (it isn't at this time).